common/key.c

```diff
@@ -89,6 +91,45 @@ KEYLIST keylist[] = {
 static int nkeylist =
     (sizeof(keylist) / sizeof(keylist[0])) - ADDITIONAL_CHARACTERS;
 
+static char* notations[] = {
+	"<C-@>",
+	"<C-a>",
+	"<C-b>",
+	"<C-c>",
+	"<C-d>",
+	"<C-e>",
+	"<C-f>",
+	"<C-g>",
+	"<C-h>",
+	/* 9 - vertical tab */
+	"",
+	/* 10 - new line */
+	"",
+	"<C-k>",
+	"<C-l>",
+	"<Enter>",
+	"<C-n>",
+	"<C-o>",
+	"<C-p>",
+	"<C-q>",
+	"<C-r>",
+	"<C-s>",
+	"<C-t>",
+	"<C-u>",
+	"<C-v>",
+	"<C-w>",
+	"<C-x>",
+	"<C-y>",
+	"<C-z>",
+	"<Esc>",
+	"<C-\\>",
+	"<C-]>",
+	"<C-^>",
+	"<C-_>",
+	/* del is 127, but we keep it on 32, as a special case */
+	"<Del>",
```

Comment:
Consider a `switch` statement instead of an array. You will find that it is being optimized into a lookup table https://godbolt.org/z/YqKf46jro (I haven't checked the correctness of this snippet)

Reply:
Oh, this is cool. Fixed.